src/librustc_mir/hair/cx/expr.rs

```diff
@@ -664,10 +664,11 @@ fn make_mirror_unadjusted<'a, 'gcx, 'tcx>(cx: &mut Cx<'a, 'gcx, 'tcx>,
                         },
                         None => offset,
                     };
-                    ExprKind::Cast { source: source.to_ref() }
+                    source.to_ref()
```

Comment:
Hmm, if I think more about this, `mk_const` should also call `.to_ref()`, to reduce the number of calls further (and to avoid having an inner `source` variable too.